### ISSUE TITLE
enable the MQTT function

### ISSUE THREAD

**Issue by charliexp** [nats-io/nats-server#2226]
I use three test machines to run nats and try to enable the MQTT function, but when simulating a server down (that is, exiting the program and running), it is easy to get the following warning:
JetStream cluster stream'$G> $MQTT_sess_KwEmus3X' has NO quorum, stalled.
And the mqtt client is also easy to be unable to connect to nats all the time.

**Comment by kozlovic** [comment]
@charliexp Would you mind sharing the configuration files of those 3 servers and what version of the NATS Server you are using? (the latest is v2.2.4).

To try to reproduce, should I shutdown one of the 3 servers and then try to connect an MQTT client, in order to get this warning, or should the client be running before stopping 1 server?

**Comment by charliexp** [comment]
### system: ubuntu 18.04
### server vesion: v2.2.4
### nats.conf:
Other server configurations are similar
```
server_name=n1-c1
listen: 0.0.0.0:4222
#jetstream: enabled
jetstream {
   store_dir: /mnt/test/data1
}
mqtt {
        listen: 0.0.0.0:1883
}
cluster {
  name: C1
  # host/port for inbound route connections from other server
  listen: 0.0.0.0:4244

  authorization {
    user: route_user
    password: pwd
    timeout: 5
  }

  routes = [
    nats-route://route_user:pwd@192.168.1.19:4244
    nats-route://route_user:pwd@192.168.1.20:4244
  ]
}
```
Run the client before stopping a server. When the server is restarted, sometimes the client can reconnect but does not receive data. At this time, if the client restarts the connection, it will fail to connect, and it has been blocked in the connection phase.
If the server is not down or restarted, these clients send and receive data normally.

**Comment by charliexp** [comment]
Simulating three service clusters with one server is the same problem.

**Comment by charliexp** [comment]
Both the message consumer side and the producter side are {qos:1}
The mqtt client cannot be connected, and it feels related to the jetstream module
```
[1843303] 2021/05/20 09:46:59.891913 [ERR] 192.168.1.2:62204 - mid:78 - unable to connect: loading session record for account "$G", session "mqtt-test-1003": JetStream system temporarily unavailable
```

**Comment by charliexp** [comment]
A node of the server restarts, the client has a reconnection function and can be reconnected, but no data is received. The client I tested is node.js, and the library is mqtt.js

code show as below:
```
var mqtt     = require('mqtt');
 var settings = {
			keepalive       : 10,
			protocolId      : 'MQTT',
			protocolVersion : 4,
			reconnectPeriod : 1000,
			connectTimeout  : 60 * 1000,
			clientId:'mqtt-test-1002',
			clean: true
		};
	
var	client = new mqtt.connect( 'mqtt://192.168.1.19:1883', settings );
		
client.on('message', function(topic, message){
    console.log( topic, message.toString() );
});
client.on('connect', function(topic, message)
{
	console.log( 'connected' );
	client.subscribe( 'a/b/#', {qos:1} );
});
```

**Comment by kozlovic** [comment]
@charliexp I am attempting to mitigate some of the issues you have seen with #2236. I think long term we need to have JetStream accept to create assets as long as there is quorum, even if the actual replicas number of servers are not running.

Regarding the issue where messages are not redelivered even when the subscription can reconnect: I think that this is more of an issue on how you set it up. You use `clean: true`, which means that when the mqtt library sends the CONNECT protocol with the clean flag, the NATS Server *has* to clean any state:
```
Spec [MQTT-3.1.2-6]: If CleanSession is set to 1, the Client and
Server MUST discard any previous Session and start a new one.
This Session lasts as long as the Network Connection. State data
associated with this Session MUST NOT be reused in any subsequent
Session.
```
Check your library, maybe it has a different event for the reconnect as opposed to event, and you should subscribe there too. Alternatively, use `clean: false` and it should work.

**Comment by charliexp** [comment]
When the clean flag is changed to true, the client can't connect and keep reporting this error:
```
[2312111] 2021/05/21 18:16:58.986255 [ERR] 192.168.1.2:62598-mid:319-unable to connect: create session stream for account "$G", session "mqtt-test-1001": timeout for request type "SC" on "$JS.API.STREAM.CREATE.$MQTT_sess_56AybFmA" (reply="$MQTT.JSA.YGhnH3VX.SC.OFC4uqLhl47wFsk6elute6")
```
When the previous clean flag is FALSE, when the server is down and restarted, the data can be pushed and the recovery time is very long.

If you switch between true and FALSE in the clean flag a few more times, you should be able to find the problem.

**Comment by charliexp** [comment]
When the clean flag is changed to true, when data is sent and received normally, one of the three servers is down and then restarted, and the subscription client connected to this server will automatically reconnect, and at this time, it will not receive another client’s send. unless the subscribed client is restarted.

**Comment by kozlovic** [comment]
I did test with the latest version of `https://github.com/eclipse/paho.mqtt.golang` using `cmd/sample/main.go` that I modified to set the proto version to 4, allow auto-reconnect, add the 3 servers in the list of possible URLs, and if the consumer was originally started with `clean: false`, then it will ultimately reconnect and receive messages sent by the publisher (that need to connect to one of the remaining servers, of course), then the consumer will receive the qos published messages.
This is using the latest 2.2.5 version. It is trued that the MQTT client will sometimes take a bit of time to reconnect because when one server is killed, depending if that server was the meta leader, or leader in one of the MQTT assets, it will take a bit for the two other servers to become leader for those assets.
Same for the publisher (if you start every time for sending a message), it may be that when you start it, it can't connect because again there is no leader for some of the mqtt assets. The fix in v2.2.5 (which is really a temporary fix until JetStream module is improved), will reduce the replicas count when a server is down and creation of the stream would result in "insufficient resources". I have created an issue for JetStream so hopefully we can address this better in a future release: https://github.com/nats-io/nats-server/issues/2239
